lib/facility_factory.rb

```diff
@@ -0,0 +1,42 @@
+class FacilityFactory
+
+  def initialize
+  end
+
+  def create_facilities(facilities)
+    if facilities[0].keys.include?(:location_1)
+      facilities.map do |facility|
+        location = JSON.parse(facility[:location_1][:human_address].gsub('\"', ''))
+        facility_details = {
+          name: facility[:title],
+          address: location.values.join(' '),
+          phone: facility[:phone_number],
+        }
+        Facility.new(facility_details)
```

Comment:
However, what you could store in the initalize method is an attribute `@or_facilities` and another for each set, that way these methods only have to run once and then all the data is stored in that attribute, instead of having these enumerables run everytime we need the data set. 

I would also have separated each of your else blocks into separate methods. Maybe a or_facilities method and a wa_facilities method. And then your if/else blocks could just call those methods depending on which logic path the code went down.

Reply:
Great job working with all the different data sets here!